### PR TITLE
Bug fixes for null exceptions, missing navigation markers,  and handle interface types.

### PR DESCRIPTION
**Pull request by srinivasankavitha**
This PR fixes
1) Null exception in DGSInputArgumentInspector
2) Handles Interface types for schema/data fetcher navigation
3) Fixes missing navigation markers for Query/Mutation types

### CODE REVIEW

File: src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java [hunk 1]
```diff
@@ -39,18 +39,22 @@ public GraphQLSchemaRegistry(Project project) {
     Optional<PsiElement> psiForSchemaType(@NotNull PsiElement psiElement, @Nullable String parentType, @Nullable String field) {
 
         TypeDefinitionRegistry registry = getRegistry(psiElement);
-        ObjectTypeDefinition type = getTypeDefinition(registry, parentType);
-        if (type != null) {
-            Optional<FieldDefinition> schemaField = type.getFieldDefinitions().stream().filter(f -> f.getName().equals(field)).findAny();
+        ObjectTypeDefinition objectType = getTypeDefinition(registry, parentType);
+        InterfaceTypeDefinition interfaceType = getInterfaceTypeDefinition(registry, parentType);
```
Comment (paulbakker) [inline src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java:43]: probalby shouldn't pre-emptively look for the interface type, but only do that if no object type was found?

File: src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java [hunk 2]
```diff
@@ -74,18 +78,29 @@ public GraphQLSchemaRegistry(Project project) {
...
         if (objectTypeExtensionDefinitions != null && !objectTypeExtensionDefinitions.isEmpty()) {
-            objectType = objectTypeExtensionDefinitions.get(0);
-        } else {
-            Optional<ObjectTypeDefinition> objectTypeDefinition = registry.getType(schemaType, ObjectTypeDefinition.class);
-            if (objectTypeDefinition.isPresent()) {
-                objectType = objectTypeDefinition.get();
-            }
+            return  objectTypeExtensionDefinitions.get(0);
+        }
+        return null;
```
Comment (paulbakker) [inline src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java:90]: Should we cahnge the API to return Optional instead? Returning null is asking for NPEs in the future

**Review by paulbakker — CHANGES_REQUESTED** [review]

File: src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java (hunk 1 above)
Comment (srinivasankavitha) [reply src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java:43]: yeah, I can change that. Was trying to make the if else branching a little less nested.

File: src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java (hunk 2 above)
Comment (srinivasankavitha) [reply src/main/java/com/netflix/dgs/plugin/services/internal/GraphQLSchemaRegistry.java:90]: makes sense, will update

**Review by paulbakker — APPROVED** [review]

**Pull request merged**